common/persistence/visibility/store/sql/query_converter_util.go

```diff
@@ -103,3 +104,15 @@ func getCoalesceCloseTimeExpr(format string) sqlparser.Expr {
 		newUnsafeSQLString(maxDatetimeValue.Format(format)),
 	)
 }
+
+func tokenizeTextQueryString(s string) []string {
+	tokens := strings.Split(s, " ")
+	n := 0
+	for i, token := range tokens {
+		if token != "" {
+			tokens[n] = tokens[i]
+			n++
+		}
+	}
+	return tokens[:n]
```

Comment:
[Clear is better than clever.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=14m35s)

I would just create another slice and copy non-empty tockend to it.